dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/Scheduler.kt

```diff
@@ -0,0 +1,89 @@
+package org.dataland.datalandbatchmanager.service
+
+import org.dataland.datalandbackend.openApiClient.api.ActuatorApi
+import org.dataland.datalandbatchmanager.gleif.CompanyUpload
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+import java.io.File
+import java.net.ConnectException
+import java.util.*
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+
+const val WAIT_TIME: Long = 5000
+
+/**
+ * Class to execute scheduled tasks, like the import of the GLEIF golden copy files
+ * @param gleifApiAccessor downloads the golden copy files from GLEIF
+ * @param gleifParser reads in the csv file from GLEIF and creates GleifCompanyInformation objects
+ * @param keycloakTokenManager manages the access tokens for authenticating against the backend API
+ */
+@Component
+class Scheduler(
```

Comment:
I don't find a class called Scheduler should not contain download/upload logic. Either rename or shift logic.